modules/seqexec/web/client/src/main/scala/seqexec/web/client/components/sequence/steps/NodAndShuffleProgress.scala

```diff
@@ -0,0 +1,165 @@
+// Copyright (c) 2016-2019 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package seqexec.web.client.components.sequence.steps
+
+import japgolly.scalajs.react._
+import japgolly.scalajs.react.vdom.html_<^._
+import react.common.implicits._
+import seqexec.web.client.model.StepItems.StepStateSnapshot
+import web.client.ReactProps
+import japgolly.scalajs.react.internal.CatsReactExt
+import seqexec.model.enum.NodAndShuffleStage
+import seqexec.web.client.components.{DividedProgress, SeqexecStyles}
+import cats.implicits._
+import seqexec.model.NodAndShuffleStep
+import seqexec.model.operations._
+import seqexec.web.client.model.ClientStatus
+import seqexec.web.client.reusability._
+
+final case class NodAndShuffleCycleProgress(state: StepStateSnapshot) extends ReactProps {
+  @inline def render: VdomElement = NodAndShuffleCycleProgress.component(this)
+}
+
+object NodAndShuffleCycleProgress extends CatsReactExt {
+  type Props = NodAndShuffleCycleProgress
+
+  implicit val propsReuse: Reusability[Props] = Reusability.derive[Props]
+
+  protected val component = ScalaComponent
+    .builder[Props]("NodAndShuffleCycleProgress")
+    .stateless
+    .render_P { p =>
+      val nsStatus = p.state.step.asInstanceOf[NodAndShuffleStep].nsStatus
+      val isInError = !p.state.isNSRunning && p.state.isNSInError
+
+      DividedProgress(
+        List.range(1, nsStatus.cycles + 1).map(_.show),
+        nsStatus.nodExposureTime.toSeconds.toInt * NodAndShuffleStage.NsSequence.length,
+        value = 0,
+        completeSectionColor = if (isInError) "red".some else "green".some,
+        ongoingSectionColor = if (isInError) "red".some else "blue".some,
+        progressCls = List(SeqexecStyles.observationProgressBar),
+        barCls = List(SeqexecStyles.observationBar),
+        labelCls = List(SeqexecStyles.observationLabel)
+      )
+    }
+    .configure(Reusability.shouldComponentUpdate)
+    .build
+}
+
+final case class NodAndShuffleNodProgress(state: StepStateSnapshot) extends ReactProps {
+  @inline def render: VdomElement = NodAndShuffleNodProgress.component(this)
+}
+
+object NodAndShuffleNodProgress extends CatsReactExt {

```

Comment:
Why are we using `CatsReactExt`?

Reply:
I think I was using `Reusability.byEq` at some point, but now I'm using `.derive[Props]`. Removing it, good catch!